src/neural/factory.h

```diff
@@ -51,13 +52,37 @@ class NetworkFactory {
     Register(const std::string& name, FactoryFunc factory, int priority = 0);
   };
 
+  // Add the network/backend parameters to the options dictionary.
+  static void PopulateOptions(OptionsParser* options);
+
   // Returns list of backend names, sorted by priority (higher priority first).
   std::vector<std::string> GetBackendsList() const;
 
   // Creates a backend given name and config.
   std::unique_ptr<Network> Create(const std::string& network, const Weights&,
                                   const OptionsDict& options);
 
+  // Helper function to load the network from the options. Returns nullptr
+  // if no network options changed since the previous call.
+  static std::unique_ptr<Network> LoadNetwork(const OptionsDict& options);
+
+  // Parameter IDs.
+  static const OptionId kWeightsId;
+  static const OptionId kBackendId;
+  static const OptionId kBackendOptionsId;
+
+  struct BackendConfiguration {
+    BackendConfiguration() {}
```

Comment:
In C++11 it's better to have `= default` instead of `{}`.
In some cases writing `{}` prevents the class from autogenerating a move constructor while `= default` doesn't.
Doesn't make a difference in this case but still a good habit.